Sources/Models/MessageStatus.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  MessageStatus.swift
+//  MessageKit
+//
+//  Created by Vlada Radchenko on 7/24/19.
+//  Copyright © 2019 MessageKit. All rights reserved.
+//
+
+import Foundation
+
+public enum MessageStatus: String, Codable {
+
```

Comment:
Shouldn't there be an `Error` or `Failed` status as well?

Reply:
You also need to update the `MockMessage` for tests to pass.

Reply:
Should be of note that this would be a breaking change to the library, it would cause everybody who implements the MessageModel protocol (so everybody) to have their implementation break. We should carefully consider if it's worth it to break everybody's code for something that MessageKit would not even make use of internally. Seems like something that could be just be specified in a concrete implementation on the user's side as opposed to requiring every consumer of the library to implement it.

Reply:
Ya I did not take that into consideration. This change wound need to be apart of a major release due to the breaking nature of it.

Reply:
Could we add a `initial` status, and extend `MessageType` protocol to return initial so it works with existing clients?

Reply:
That could work but IMHO, anything that's not used by the library itself probably doesn't need to be in the `MessageType` protocol. Since it's just a protocol, clients of the library already have the flexibility to add a `MessageStatus` field to their message struct/class if they desire.
````swift
struct Message: MessageType {
    // ... MessageType implementation
    let status: MessageStatus
}
````
This issue has come up before (#807) and it seems like Nathan and Steven believed this to be a more client side issue.

Reply:
Makes sense. @Vlada31R what do you think? We would probably just close this and approach as kinoroy recommended above